### PR TITLE
Fixes

### LINKED ISSUE #29: increase size of architecture diagram

### PR DESCRIPTION
**Pull request by JayjeetAtGithub**
fixes #29, #28, #27

### CODE REVIEW

**Comment by JayjeetAtGithub** [comment]
> expand listing caption.

@ivotron Can you please help me on this ?

File: paper.md [hunk 1]
```diff
@@ -81,8 +81,12 @@ Let us take a relatively simple scenario where users have a list of single-purpo
...
-
-**TODO**: describe what we do **_without_** Popper, so that it becomes evident why we need to have Popper.
+In practice, when developers work following the container-native paradigm they end up interactively executing multiple Docker commands to build containers, compile code, test applications, deploy software, etc. 
+Keeping track of which commands were executed, in which order, and which flags were passed to each, can quickly become unmanageable, difficult to document, error prone, and hard to reproduce.
+The goal of Popper is to bring order to this chaotic scenario by providing a framework for clearly and explicitly defining container-native tasks.
+With Popper, users need to find or build Docker images containing the required dependencies, write a workflow in YAML, and execute a `popper run`.
+Popper will automate the process by running the scripts inside the specified containers as individual steps in the prescribed order.
+The workflow file can be shared through GitHub and the experiments can be reproduced in different computing environments with a single command.
```
Comment (ivotron) [inline paper.md:89]: these three lines can be omitted as this is introduced later. The goal for this section is to talk about the problem that is being solved. The other thing we can talk about is that, if we want to run the same manual workflow but on another environment, the set of commands we execute end up being different. For example, singularity manages images differently (we can mention its cache, the flat image format, etc).

we can also expando on how running the same workflow but on a k8s or slurm means we will need to do distinct things

File: paper.md [hunk 2]
```diff
@@ -105,7 +109,7 @@ The key feature that differentiates it from Docker is that it can be used in non
 It also provides an abstraction that enables using container images from different image registries interchangeably like Docker Hub, Singularity Hub, and Sylabs Cloud.
 These features make Singularity increasingly useful in areas of Machine learning, Deep learning, and other data-intensive applications where the workloads benefit from the HPC support of it.
 
-![DOT diagram of a Popper workflow DAG](./figures/wf.pdf){#fig:casestudy}
+![DOT diagram of a Popper workflow DAG](./figures/wf.pdf){#fig:casestudy .center height=35%}
```
Comment (ivotron) [inline paper.md:112]: this figure is referenced _after_ the code listing, so it flows better if the figure appears after the listing

File: paper.md [hunk 3]
```diff
@@ -138,25 +142,20 @@ Several hosted CI services like Travis, Circle, and Jenkins make continuous inte
 YAML [@ben2009yaml] is a human-readable data-serialization language. 
 It is commonly used in writing configuration files and in applications where data is stored or transmitted. 
 Due to its simplicity and wide adoption [@yaml_wide_adoption], we chose YAML for defining popper workflows and for specifying the configuration for the execution engine. 
-An example of a popper workflow is shown below.
+An example popper workflow is shown in @Lst:wf-example.
```
Comment (ivotron) [inline paper.md:145]: For the "expand listing caption" todo item, we can explain what each step is doing in more detail.

File: paper.md [hunk 4]
```diff
@@ -227,26 +226,26 @@ For example, a workflow developed on the local machine can be run on an HPC clus
 The configuration file can be passed through the CLI interface and can be shared among different workflows.
 It can either be created by users or provided by system administrators.
 
-![Architecture of the Popper workflow engine](./figures/architecture.pdf){#fig:arch}
+![Architecture of the Popper workflow engine](./figures/architecture_med.pdf){#fig:arch}
```
Comment (ivotron) [inline paper.md:229]: nice!

File: paper.md [hunk 5]
```diff
@@ -227,26 +226,26 @@ For example, a workflow developed on the local machine can be run on an HPC clus
...
+To set up CI for a project using Popper, it is required to generate a CI configuration file, push the project to GitHub and enable the repository on the CI provider.
 Using CI with Popper workflows enhances the reproducibility guarantees as continuous validation helps to keep a check on various breaking changes like outdated dependencies, broken links, deleted Docker images, etc.
 Another benefit of using CI with Popper is that even without changes, jobs can be configured so that they run periodically (e.g. once a week), to ensure that they are in a healthy state.
 
 # Case Study {#sec:casestudy}
 
 In this section, we present three case studies demonstrating how the Popper workflow engine allows reproducing and scaling workflows in different computing environments.
-We analyzed the ML-based system benchmarking project MLPerf [@mattson2019mlperf] based on the reproducibility related issues that get frequently opened on its Github repository 
+We analyzed the ML-based system benchmarking project MLPerf [@mattson2019mlperf] based on the reproducibility related issues that get frequently opened on its GitHub repository
```
Comment (ivotron) [inline paper.md:242]: we can also add a link to the repository

Comment (ivotron) [reply paper.md:242]: also, we can reiterate how Popper is solving this in the Results section, as this is a central point.

File: paper.md [hunk 6]
```diff
@@ -374,9 +373,9 @@ This case study showcases the benefits of using Popper: having portable workflow
 
 The adjustments that users need to make to reproduce workflows on Kubernetes and Slurm is described below.
 
-1. To run workflows on Kubernetes clusters, users need to pass some configuration options through a YAML file with contents similar to the one shown below.
+1. To run workflows on Kubernetes clusters, users need to pass some configuration options through a YAML file with contents similar to the one shown in @Lst:kubernetes.
 
-```yaml
+```{#lst:kubernetes .yaml caption="Config file for running on Kubernetes."}
```
Comment (ivotron) [inline paper.md:378]: based on the latest changes we introduced and the ones we plan to work on, registry info won't be necessary anymore, so we can remove it. Only namespace and volume_size will be needed. We can make a note that this is only required if the user is not OK with the defaults

**Review by ivotron — CHANGES_REQUESTED** [review]

File: paper.md [hunk 7]
```diff
@@ -385,9 +384,9 @@ resource_manager:
     namespace: mynamespace
 ```
 
-2. Similarly, for running on Slurm, users need to specify few configuration options like the number of nodes to use for running the job concurrently, the number of CPUs to allocate to each task, the worker nodes to use, etc.
+2. Similarly, for running on Slurm, users need to specify few configuration options like the number of nodes to use for running the job concurrently, the number of CPUs to allocate to each task, the worker nodes to use, etc. as shown in @Lst:slurm.
 
-```yaml
+```{#lst:slurm .yaml caption="Config file for running on Slurm."}
```
[the remark this answers is not in the document]
Comment (ivotron) [reply paper.md:389]: in this way we don't need a configuration file

[the remark this answers is not in the document]
Comment (JayjeetAtGithub) [reply paper.md:389]: Yeah we can. But I believe currently we do not have that support. Should I leave the config as it is now ?

[the remark this answers is not in the document]
Comment (ivotron) [reply paper.md:389]: yeah, you are right, we can leave it as is, as this is not currently supported. Maybe if paper gets accepted and we have the feature by then we can mention that.

[the remark this answers is not in the document]
Comment (ivotron) [reply paper.md:389]: I added this https://github.com/getpopper/popper/issues/920

[the remark this answers is not in the document]
Comment (JayjeetAtGithub) [reply paper.md:389]: Sounds Good !

**Pull request merged**